hmy/tracer.go

```diff
@@ -342,10 +342,84 @@ func (hmy *Harmony) TraceChain(ctx context.Context, start, end *types.Block, con
 	return sub, nil
 }
 
+// same as TraceBlock, but only use 1 thread
+func (hmy *Harmony) traceBlockNoThread(ctx context.Context, block *types.Block, config *TraceConfig) ([]*TxTraceResult, error) {
```

Comment:
can you add instrumentation to keep track of the execution time for comparison?

Reply:
It doesn't reduce execution time, it just reduces thread usage.
another bug fixed is: if a client uses a batch RPC to trace some blocks, and the connection broke during the execution of these blocks, the Trace module still completes all the tasks. this wastes a lot of CPUs.